src/main/java/fr/esgi/al5_2/Tayarim/services/AdministrateurService.java

```diff
@@ -0,0 +1,171 @@
+import fr.esgi.al5_2.Tayarim.entities.Administrateur;
+import fr.esgi.al5_2.Tayarim.entities.Proprietaire;
+import fr.esgi.al5_2.Tayarim.exceptions.*;
+import fr.esgi.al5_2.Tayarim.mappers.AdministrateurMapper;
+import fr.esgi.al5_2.Tayarim.mappers.ProprietaireMapper;
+import fr.esgi.al5_2.Tayarim.repositories.AdministrateurRepository;
+import fr.esgi.al5_2.Tayarim.repositories.ProprietaireRepository;
+import lombok.NonNull;
+import org.mindrot.jbcrypt.BCrypt;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = true)
+public class AdministrateurService {
+
+  private final AdministrateurRepository administrateurRepository;
+
+  public AdministrateurService(AdministrateurRepository administrateurRepository) {
+    this.administrateurRepository = administrateurRepository;
+  }
+
+  @Transactional
+  public AdministrateurDTO creerAdministrateur(
+      @NonNull AdministrateurCreationDTO administrateurCreationDTO) {
+    if (administrateurRepository.findFirstByEmail(administrateurCreationDTO.getEmail())
+        .isPresent()) {
+      throw new AdministrateurEmailAlreadyExistException();
+    }
+
+    String numTel = administrateurCreationDTO.getNumTel();
+    numTel = numTel.replaceAll(" ", "");
+    if (administrateurRepository.findFirstByNumTel(numTel).isPresent()) {
+      throw new AdministrateurNumTelAlreadyExistException();
+    }
+    administrateurCreationDTO.setNumTel(numTel);
+
+    String hashedPassword = hashPassword(administrateurCreationDTO.getMotDePasse());
+
+    administrateurCreationDTO.setMotDePasse(hashedPassword);
+
+    Administrateur administrateur = AdministrateurMapper.creationDtoToEntity(
+        administrateurCreationDTO);
+
+    return AdministrateurMapper.entityToDto(administrateurRepository.save(administrateur));
+  }
+
+  public List<AdministrateurDTO> getAdministrateur() {
+    return AdministrateurMapper.entityListToDtoList(administrateurRepository.findAll());
+  }
+
+  public AdministrateurDTO getAdministrateurById(@NonNull Long id) {
+
+    Optional<Administrateur> optionalAdministrateur = administrateurRepository.findById(id);
+    if (optionalAdministrateur.isEmpty()) {
+      throw new AdministrateurNotFoundException();
+    }
+    return AdministrateurMapper.entityToDto(optionalAdministrateur.get());
+  }
+
+  public AdministrateurDTO getAdministrateurByEmail(@NonNull String email) {
+    Optional<Administrateur> optionalAdministrateur = administrateurRepository.findFirstByEmail(
+        email);
+    if (optionalAdministrateur.isEmpty()) {
+      throw new AdministrateurNotFoundException();
+    }
+
+    return AdministrateurMapper.entityToDto(optionalAdministrateur.get());
+  }
+
+  @Transactional
+  public AdministrateurDTO updateAdministrateur(@NonNull Long id,
+      @NonNull AdministrateurUpdateDTO administrateurUpdateDTO) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'administrateurUpdateDTO' must contain no more than '1' consecutive capital letters.